native/src/modules/app/LocationPermissionManager.js

```diff
@@ -2,15 +2,22 @@
 
 import { check, PERMISSIONS, request, RESULTS } from 'react-native-permissions'
 import { Platform } from 'react-native'
+import buildConfig from './constants/buildConfig'
 
 export const checkLocationPermission = async (): Promise<RESULTS> => {
+  if (buildConfig().featureFlags.fixedCity) {
+    throw Error('Can not check location permission when city is fixed')
+  }
```

Comment:
I think these methods are still called in the PermissionSnackbarContainer even if there is a fixed city. So I guess this may crash the app. Did you check that the app does not crash?

Reply:
I think you may have to check the `updateSettingsAndPermissions` method in `PermissionSnackbarContainer`

Reply:
Another idea: We could also extend the `RESULTS` type from react-native-permissions instead of throwing an exception

Reply:
I think this does not crash the app because this code if not reachable if `buildConfig().featureFlags.fixedCity`

Reply:
> I think this does not crash the app because this code if not reachable if `buildConfig().featureFlags.fixedCity`

Actually I think it is since the PermissionSnackbarContainer is still used :D But I may be wrong.

Reply:
You were both right 'buildConfig().featureFlags.fixedCity' prevented the PermissionSnackbar from showing, but the app crashed anyways because updateSettingsAndPermissions of PermissionSnackbarContainer was calling checkLocationPermission.

Anyway, I fixed it as suggested by Max by extending the RESULTS type.